algo_arena_and_results

idraper opened the pull request:
There are two files in this PR.

run_arena.py takes algos as inputs and then runs every combination of games for a set batch_size.

get_results.py is the beginning of a replay analyzer and has the ability to display various data from replay files.

Let me know if you have any comments, suggestions, or required edits :).

idraper commented:
Alright, I created a header for each of those files. I also included a copyright so people know they can do whatever they want with it and also a short description for each.

I also added `#!/usr/bin/env python` for Unix users, since I forgot to put that in before. I made some small changes to the instructions for each to make them slightly more clear and up-to-date.

876584635678890 commented on scripts/contributions/get_results.py:
```diff
@@ -0,0 +1,737 @@
+	numRows, numCols = 0,0
+	fig, ax, arg = None, None, None
+	pos = (0,0)
+	emptyPlots = []
```
Use of camel case, which is not idiomatic.

876584635678890 commented on scripts/contributions/get_results.py:
```diff
@@ -0,0 +1,737 @@
+	pos = (0,0)
+	emptyPlots = []
+
+	verbose_options = ['health', 'bits', 'cores', 'cores_spent', 'bits_spent', 'cores_on_board']
```
Inconsistent use of camel case and underscores.

876584635678890 commented:
Really like the idea of it.  
I did some testing and it did not work for me because my replays come without names for some reason, which results in crashing.
My review is not mandatory at all, but it is an improvement I would make.

idraper commented:
Thanks for helping! I totally was switching between c++ and python naming conventions without even noticing, so thanks for pointing that out. I've gone through and changed everything (I hope) to be python in nature.

When you say you don't have name, do you mean the replay file doesn't have a name or the algo doesn't? If it is the algo, do you have two algos that have the same name?

876584635678890 commented on scripts/contributions/get_results.py:
```diff
@@ -0,0 +1,736 @@
+If you have any suggestions/questions let me know :) - @Isaac
+'''
+
+pltInstalled = False
```
```suggestion
plt_installed = False
```

876584635678890 commented on scripts/contributions/get_results.py:
```diff
@@ -0,0 +1,736 @@
+
+try:
+	import matplotlib.pyplot as plt
+	pltInstalled = True
```
```suggestion
	plt_installed = True
```

876584635678890 commented on scripts/contributions/get_results.py:
```diff
@@ -0,0 +1,736 @@
+	pltInstalled = True
+except ImportError:
+	try:
+		usrIn = input('Matplotlib not found.\nWould you like this program to try and install matplotlib? (y/n) ')
```
```suggestion
		usr_in = input('Matplotlib not found.\nWould you like this program to try and install matplotlib? (y/n) ')
```

876584635678890 commented on scripts/contributions/get_results.py:
```diff
@@ -0,0 +1,736 @@
+except ImportError:
+	try:
+		usrIn = input('Matplotlib not found.\nWould you like this program to try and install matplotlib? (y/n) ')
+		if usrIn.lower() == 'y' or usrIn.lower() == 'yes':
```
```suggestion
		if usr_in.lower() == 'y' or usr_in.lower() == 'yes':
```

876584635678890 commented on scripts/contributions/get_results.py:
```diff
@@ -0,0 +1,736 @@
+			subprocess.run(['python', '-m', 'pip', 'install', 'matplotlib'])
+
+			import matplotlib.pyplot as plt
+			pltInstalled = True
```
```suggestion
			plt_installed = True
```

876584635678890 commented on scripts/contributions/get_results.py:
```diff
@@ -0,0 +1,736 @@
+			pltInstalled = True
+			sys.stderr.write("\n\n")
+	except:
+		pltInstalled = False
```
```suggestion
		plt_installed = False
```

876584635678890 commented on scripts/contributions/get_results.py:
```diff
@@ -0,0 +1,736 @@
+
+	# check to see if matplotlib is installed
+	graphing_enabled = True if len(args['graph']) > 0 else False
+	if graphing_enabled and not pltInstalled:
```
```suggestion
	if graphing_enabled and not plt_installed:
```

876584635678890 commented on scripts/contributions/get_results.py:
```diff
@@ -0,0 +1,736 @@
+		fig_size = [fig_size[0] * c, fig_size[1] * r]
+		if fig_size[0] > max_x: fig_size[0] = max_x
+		if fig_size[1] > max_y: fig_size[1] = max_y
+		plt.rcParams["figure.figsize"] = fig_size
```
Not sure about this one.

idraper replied:
`rcParams` is specific to matplotlib and cannot be changed.

876584635678890 commented on scripts/contributions/get_results.py:
```diff
@@ -0,0 +1,736 @@
+		self.turns = {}
+		self.valid_turns = []
+
+		self.loadData()				# handles loading all the data from file into python variables
```
```suggestion
		self.load_data()				# handles loading all the data from file into python variables
```

876584635678890 commented on scripts/contributions/get_results.py:
```diff
@@ -0,0 +1,736 @@
+	def __repr__(self):
+		return self.__toString()
+
+	def loadData(self):
```
```suggestion
	def load_data(self):
```

876584635678890 commented on scripts/contributions/get_results.py:
```diff
@@ -0,0 +1,736 @@
+		if type(other) == str:
+			return self.name == other
+		return self.name == other.name
+	def __toString(self):
```
```suggestion
	def __string(self):
```

876584635678890 commented on scripts/contributions/get_results.py:
```diff
@@ -0,0 +1,736 @@
+	def __toString(self):
+		return self.name
+	def __str__(self):
+		return self.__toString()
```
```suggestion
		return self.__string()
```

876584635678890 commented on scripts/contributions/get_results.py:
```diff
@@ -0,0 +1,736 @@
+	def __str__(self):
+		return self.__toString()
+	def __repr__(self):
+		return self.__toString()
```
```suggestion
		return self.__string()
```

idraper commented:
Thanks for the input. I changed all the additional naming errors you noticed. I also fixed the `'name'` bug you were experiencing.

I'll look into the matplotlib error now.

876584635678890 commented:
I fixed it, needed to install `tkinter`.

idraper commented:
@876584635678890  really? How does that occur? Looking at the matplotlib dependencies:
Requires-Python: >=3.5
Requires-Dist: numpy (>=1.10.0)
Requires-Dist: cycler (>=0.10)
Requires-Dist: kiwisolver (>=1.0.1)
Requires-Dist: pyparsing (!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1)
Requires-Dist: python-dateutil (>=2.1)

876584635678890 commented:
https://stackoverflow.com/q/36327134/6509751

Also, plotting works fine, but only when using `-v` as well. Otherwise it will only plot total wins.

idraper commented:
That is intended use. If you run more than one game it defaults to "summary" view, which is only information about the aggregate of matches. The -v flag forces it to graph more information. This is information you don't necessarily care about if you are running it on multiple games.

skelly1324 commented:
Fantastic stuff here, this is great.
